cmd/podman/containers/cp.go

```diff
@@ -113,6 +116,82 @@ func doCopy(funcA func() error, funcB func() error) error {
 	return errorhandling.JoinErrors(copyErrors)
 }
 
+func copyContainerToContainer(fromContainer string, sourcePath string, toContainer string, destPath string) error {
+	if err := containerMustExist(fromContainer); err != nil {
+		return err
+	}
+
+	if err := containerMustExist(toContainer); err != nil {
+		return err
+	}
+
+	fromContainerInfo, err := registry.ContainerEngine().ContainerStat(registry.GetContext(), fromContainer, sourcePath)
+	if err != nil {
+		return errors.Wrapf(err, "%q could not be found on container %s", sourcePath, fromContainer)
+	}
+
+	var toContainerBaseName string
```

Comment:
can you change to `toContainerBaseName := ""` removes an unnecessary line.

Reply:
Also, I am seeing you are getting an ineffectual assignment error here. Where are you using `toContainerBaseName` after setting it?

Reply:
> can you change to toContainerBaseName := "" removes an unnecessary line.

sure, will make the changes once I figure everything out 

> Also, I am seeing you are getting an ineffectual assignment error here. Where are you using toContainerBaseName after setting it?

I am not completely sure tbh, I had to do _ = toContainerBaseName to get rid of that unused variable error

Reply:
ok @infiniteregrets you're getting that error because you set the base name and never actually use it. the `_ =` is just a clever way to trick golang. You should use that name somewhere in your copy code.

Reply:
@cdoern I am using it? 
```
if toContainerInfo != nil {
			toContainerBaseName = filepath.Base(toContainerInfo.LinkTarget)
		} else {
			toContainerBaseName = filepath.Base(destPath)
		}
```		
This looks like a shadowing issue? Im not sure but I'll try to figure it. Thanks!